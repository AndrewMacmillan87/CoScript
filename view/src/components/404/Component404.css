.four0four-body {
    background: url("../../img/vintage-books.jpg") center center no-repeat;
    background-size: cover;
    height: 100vh;
    width: 100%;
}

.four0four-container {
    text-align: center;
    padding: 30vh 0;
    width: 100%;
    color: #fff;
}

.four0four-header {
    font-family: "Dancing Script", cursive;
    margin: 0;
    font-size: 4rem;
}

.four0four-text {
    display: block;
    margin: 10px 0;
    font-family: "Lato", sans-serif;
    font-size: 1.6rem;
    padding: 5px;
}
