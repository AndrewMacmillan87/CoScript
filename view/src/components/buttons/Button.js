import React from "react";
import PropTypes from "prop-types";

import "./buttons.css";

const Button = ({ onClick, id, classes, value }) => {
    return (
        <button onClick={onClick} id={id} className={classes} type='button'>
            <span className='btn-text'>
                <b>{value}</b>
            </span>
        </button>
    );
};

Button.propTypes = {
    onClick: PropTypes.func,
    classes: PropTypes.string,
    id: PropTypes.string,
    value: PropTypes.string
};

export default Button;
