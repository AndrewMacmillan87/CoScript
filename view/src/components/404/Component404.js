import React from "react";
import { Link } from "react-router-dom";
import Button from "../buttons/Button";

import "./Component404.css";

const Component404 = () => {
    return (
        <section className='four0four-body'>
            <div className='four0four-container'>
                <h1 className='four0four-header'>Ooops, 404...</h1>
                <p className='four0four-text'>
                    It looks like the page you were looking for isn't here :(
                </p>
                <Link to='/'>
                    <Button classes='small-btn' value='Home Page' />
                </Link>
            </div>
        </section>
    );
};

export default Component404;
