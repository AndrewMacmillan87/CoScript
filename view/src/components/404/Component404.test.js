import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Provider } from "react-redux";
import { reduxStore } from "../../redux/store/reduxStore";

import Component404 from "./Component404";
import Button from "../buttons/Button";

describe("404 component", () => {
    test("renders without crashing", () => {
        global.scrollTo = jest.fn();
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={reduxStore}>
                <Router>
                    <Component404 />
                </Router>
            </Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders correctly", () => {
        global.scrollTo = jest.fn();
        const tree = renderer
            .create(
                <Provider store={reduxStore}>
                    <Router>
                        <Component404 />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render a section element with a className of four0four-body", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Component404 />
                </Router>
            </Provider>
        );

        expect(wrapper.find("section").length).toEqual(1);
        expect(wrapper.find({ className: "four0four-body" }).length).toEqual(1);
    });

    it("should render 1 div element with a className of four0four-container", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Component404 />
                </Router>
            </Provider>
        );

        expect(wrapper.find("div").length).toEqual(1);
        expect(
            wrapper.find({ className: "four0four-container" }).length
        ).toEqual(1);
    });

    it("should render 1 h1 element with a className of four0four-header", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Component404 />
                </Router>
            </Provider>
        );

        expect(wrapper.find("h1").length).toEqual(1);
        expect(wrapper.find({ className: "four0four-header" }).length).toEqual(
            1
        );
    });

    it("should render 1 p element with a className of four0four-text", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Component404 />
                </Router>
            </Provider>
        );

        expect(wrapper.find("p").length).toEqual(1);
        expect(wrapper.find({ className: "four0four-text" }).length).toEqual(1);
    });

    it("should render 1 Button component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Component404 />
                </Router>
            </Provider>
        );

        expect(wrapper.find(Button).length).toEqual(1);
    });
});
