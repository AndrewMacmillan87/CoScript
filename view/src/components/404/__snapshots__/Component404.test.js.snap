// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 component renders correctly 1`] = `
<section
  className="four0four-body"
>
  <div
    className="four0four-container"
  >
    <h1
      className="four0four-header"
    >
      Ooops, 404...
    </h1>
    <p
      className="four0four-text"
    >
      It looks like the page you were looking for isn't here :(
    </p>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="small-btn"
        type="button"
      >
        <span
          className="btn-text"
        >
          <b>
            Home Page
          </b>
        </span>
      </button>
    </a>
  </div>
</section>
`;
